iothub/device/src/DeviceClient.ConventionBasedOperations.cs

```diff
@@ -37,5 +36,31 @@ public Task SendTelemetryAsync(TelemetryMessage telemetryMessage, CancellationTo
             Func<CommandRequest, object, Task<CommandResponse>> callback, object userContext,
             CancellationToken cancellationToken = default)
             => InternalClient.SubscribeToCommandsAsync(callback, userContext, cancellationToken);
+
+        /// <summary>
+        /// Retrieve the device properties.
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+        /// <returns>The device properties.</returns>
+        public Task<ClientProperties> GetClientPropertiesAsync(CancellationToken cancellationToken = default)
+            => InternalClient.GetClientPropertiesAsync(cancellationToken);
+
+        /// <summary>
+        /// Update properties.
```

Comment:
Are there any additional details you can provide here about how this API works? Is it a patch or a replace style update?

Reply:
I'll add a note calling out that this is a patch operation and that we don't need to send the entire twin object.

Reply:
well, not twin, properties